### ISSUE TITLE
Table: NVDA reads whole table data at once upon navigating to it via Tab key

### ISSUE THREAD

**Issue by oyashyna** [cengage/react-magma#1797]
**Describe the bug**
NVDA screen reader reads hole table data (column headers, rows and cells values) at once upon navigating to it via Tab key

**To Reproduce**
1. Go to React Magma Docs
2. Open Table -> Basic Usage
3. Turn on NVDA screen reader
4. Move to table example via Tab key
5. Verify NVDA screen reader reads hole table data (column headers, rows and cells values) at once

**Expected behavior**
Screen reader should read only the table name. Headers/cells should be read when user navigates to them

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser:  Chrome, v. 137, Firefox 139
 - Screen reader: NVDA

**Additional context**
May be connected to https://github.com/cengage/react-magma/issues/1769

**Comment by nikitaorliak-cengage** [comment]
Needs investigation

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
- V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/table--default
- V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/table--default

~~On Windows + Chrome + NVDA, screenreader announces the whole table.~~
<img width="487" height="775" alt="Image" src="https://github.com/user-attachments/assets/2bac1a7c-23e8-4357-a609-f359cfaae28c" />

~~On Windows + Firefox + NVDA and on MacOS + VO everything works as expected.~~

**Update:**
In Browse Mode Windows + Nvda + Chrome/Firefox reads the whole table. In Focus mode, it reads only the header (as expected). In this case, we have 2 possible solutions:
1. Force screenreaders to change their focus behavior on scrollable tables (e.g., use `role="application"` or something else)
2. Leave this up to the users/devs. If they are familiar with the features of their focus/browse modes, they should understand the consequences of focusing on a table in Browse Mode
